CommuniTEAM/CommuniTEA
# [TEA-54] Backend Authentication

### In this PR, I'm adding/changing the following:

- Backend authentication :)
  - Actual detailed changelog tbd

### Please do the following to test my code:

1. Update your local `.env` file with the changes in `.env.sample`
2. Delete docker containers and `communitea-db` volume and recreate/rebuild
3. Navigate to localhost:8000/docs and check out the updated UI
4. Create a location with the city name "string" and any state abbreviation **in Beekeeper**:
```
insert into locations_cities
values ('1742bc8e-a057-11ee-8c90-0242ac120002', 'string', 'CA');
```
6. Create a user in the Swagger UI (for example purposes, it will automatically grab the city named "string" as the new user's location)
7. Try to make a location in the Swagger UI (it should fail as you are not an admin user)
8. Create a new user with the username "admin"
9. Update your new user's role to admin **in Beekeeper**
```
update users
set role = 'admin'
where username = 'admin'
```
10. While logged in as admin in the Swagger UI, try to create a new location -- it should work!
11. Try logging out
12. Try to make another location -- it should fail as you are not authenticated.


### Notes:

I will update this PR with more detailed docs and information ASAP, but for now just test and lmk how it goes!

[Related Jira Ticket](https://communiteam.atlassian.net/browse/TEA-54)
